state/protocol/badger/mutator.go

```diff
@@ -641,14 +668,12 @@ func (m *FollowerState) Finalize(ctx context.Context, blockID flow.Identifier) e
 
 	// We update metrics and emit protocol events for epoch state changes when
 	// the block corresponding to the state change is finalized
-	epochStatus, err := m.epoch.statuses.ByBlockID(blockID)
+	protocolState, err := m.protocolStateReader.AtBlockID(blockID)
```

Comment:
I would suggest to emphasize that this is a snapshot (at a certain block) to reduce naming ambiguity in the broader implementation

```suggestion
	psSnapshot, err := m.protocolStateReader.AtBlockID(blockID)
```